protocols/lightning-yaelbot.md

```diff
@@ -0,0 +1,32 @@
+# Lightning, Yet An Extra Little Bit Of Trust
+
+YAELBOT allows opening a channel with a trusted peer simply by setting up
+a multi-sig 2of2 (but not just a one-time address, but rather an unlimited
+[Electrum Bitcoin Wallet-like][e2of2] wallet which allows movement at
+any time) and then informing the node about a mutually-signed state
+of channel.
```

Comment:
Among whom is the multi-sig 2of2? Between you and a new trusted peer, watchtower-like? Or with the other channel counterparty?

Reply:
No watchtowers would be needed in YAELBOT channels.

`A`lice and `B`ob set up a YAELBOT channel. None of them can close it unilaterally, only after a mutual agreement.

`A` will never give their private key to `B` and vice-versa, but `A` may give the private key to their family member and instructions about how to contact `B` to another family member. `B` knows that until `A` lives, no request from anyone would be co-signed by `B` (even if everyone except `B` had `A`'s private key). And vice versa. The the current state and history of channel's balance can be kept any way (git repository, piece of paper, …), whereas only the current state electronically signed by both parties (TweetNaCl-like) is needed to route payments from other sides (including LN).

Emergency cases can be pre-negotiated freely and all the backups informed about possible states.

I am no cryptographer, just a fan, but this either can be done or I'll be proven wrong. I think that on technical level it is simpler than current Lightning Network implementations because most of the tasks are off-loaded to people and their agreement with only requirement that they will sign and prove something that other nodes can verify. When the any of the multi-sig UTXOs comprising the channel move without announcing the channel state change, the channel is considered gone. New funds can be easily added by any party (and even donors) and the channel state will be updated manually after the fact. All nodes just extend or shrink the size of the channel in their view once the channel state is updated.

Reply:
>A will never give their private key to B and vice-versa, but A may give the private key to their family member and instructions about how to contact B to another family member. B knows that until A lives, no request from anyone would be co-signed by B (even if everyone except B had A's private key). And vice versa. The current state and history of channel's balance can be kept any way (git repository, piece of paper, …), whereas only the current state electronically signed by both parties (TweetNaCl-like) is needed to route payments from other sides (including LN).

This idea of manual management of the channel is cool, and I like it. However, there are a couple of points that is not clear to me that are summarized below

- What happens if I miss the control of the lightning network node? with the actual state of the art I can not recover the channel if I miss my database, and I can't use an old backup of the database because of the rules of the lightning network punish me. Do you think that in this case, we (as a User) need to have a little bit of automatic stuff that saves all the bad stuff that a human can do?
- Another point that I think about is if we have a channel connected with the world, this means that I can have a lot of transactions that will drain the capacity of the channel in the time. In this case, we need to close the channel manually each time that the channel is drained from other payments.

A useful task where an N-of-M script it is cool is when I decided to centralize a node lightning network in one server and this server is shared between different people (like mobile user) the script N-of-M help to have an agreement between all participant of this centralized lightning node when I make an operation on-chain.

What do you think?

PS: Not sure if the N-of-M script exist already in some implementation.
PSS: I'm no a people with a deep concept of Bitcoin Script and an expert of Lightning Network, so this is only an opinion and a couple of ideas that I had when I have

Reply:
Thank you for interesting comments and ideas, @vincenzopalazzo!

> * What happens if I miss the control of the lightning network node? with the actual state of the art I can not recover the channel if I miss my database, and I can't use an old backup of the database because of the rules of the lightning network punish me. Do you think that in this case, we (as a User) need to have a little bit of automatic stuff that saves all the bad stuff that a human can do?

The keys which will be used to set-up the channel will be preferably offline, backed up physically (paper, metal, …). The private keys used by both sides to sign the current balance of the channel will have no right to close the channel.

The idea is not removing anything from the current LN implementation, just adding manually-operated (MO) channels. Closing of such channels will be based on agreement only. The details can be settled in _any way_ both parties agree to.

> * Another point that I think about is if we have a channel connected with the world, this means that I can have a lot of transactions that will drain the capacity of the channel in the time. In this case, we need to close the channel manually each time that the channel is drained from other payments.

Any (LN or MO) channel can be re-balanced without closing the channel itself. Example:

_Me and you have a Lightning Network channel and all the funds are currently on your side. I send an on-chain Bitcoin transaction to you (or buy you an icecream, or anything else we agree upon) and you do a Lightning Network transaction\* to me in order to re-balance the channel. No need to close it._

\* `pay` an invoice or do a `keysend`

With MO channel the only difference will be that we can be sure the channel will never be closed without a willful agreement and manual action.

> A useful task where an N-of-M script it is cool is when I decided to centralize a node lightning network in one server and this server is shared between different people (like mobile user) the script N-of-M help to have an agreement between all participant of this centralized lightning node when I make an operation on-chain.

MO channel funding would be possible in N-of-M arrangements. But practically it would still just be a channel between two nodes.

> PS: Not sure if the N-of-M script exist already in some implementation.

N-of-M is implemented at least in [Electrum](https://electrum.org/#home) and [Caravan](https://unchained-capital.github.io/caravan/#/) that I know of. Probably in many more wallets.

> PSS: I'm no a people with a deep concept of Bitcoin Script and an expert of Lightning Network, so this is only an opinion and a couple of ideas that I had when I have

Explore the idea if it interests you. But keep in mind that anything more than two ends of a channel may lead to re-inventing Bitcoin.